packet/flatcar-linux/kubernetes/cl/controller.yaml.tmpl

```diff
@@ -282,6 +283,15 @@ storage:
           kind: KubeletConfiguration
           cgroupDriver: "$${docker_cgroup_driver}"
           EOF
+    - path: /opt/etcd-private-interface
+      filesystem: root
+      mode: 0744
+      contents:
+        inline: |
+          #!/bin/bash
+          set -e
+          ipaddress=$(curl metadata.packet.net/metadata | jq -r ".network.addresses[] | select(.public == false) | .address")
+          sed -i "s|REPLACE_WITH_PRIVATE_IP|$${ipaddress}|g" /etc/systemd/system/etcd-member.service.d/40-etcd-cluster.conf
```

Comment:
I would detect if the command was run already and then not touch the file. But if the file was changed, then you need to run `systemctl daemon-reload` to apply the change and also you have to fail the current execution because the service will otherwise use the old environment.

Reply:
Added following:

```
          systemctl daemon-reload
          if ! test -f /tmp/etcd-private-interface; then
            touch /tmp/etcd-private-interface
            exit 1
          fi
```

Reply:
Nvm

Reply:
done